app/src/components/report/ReportEdit.tsx

```diff
@@ -1,73 +1,126 @@
 import { IReport } from '@/models';
-import { useCallback } from 'react';
-import Button from 'react-bootstrap/Button';
-import Form from 'react-bootstrap/Form';
-import Spinner from 'react-bootstrap/Spinner';
-import Stack from 'react-bootstrap/Stack';
+import { Spinner, Stack, Button, Form } from 'react-bootstrap';
 import ButtonLink from '../ButtonLink';
-import { ReportCategories, StatusOfReport } from '@/models';
+import { ReportCategories } from '@/models';
+import { useFormik, FormikHelpers } from 'formik';
+import { isUndefined } from '@/utils';
+import { VALIDATION_SCHEMA } from '../../utils/validation/ValidSchema';
+
+//Initial values for report form
+type NewReport = Omit<IReport, 'reportId'>;
+type InitialValuesType = NewReport | IReport;
+
+const getInitialReport = (): NewReport => ({
+    name: '',
+    emailAddress: '',
+    phoneNumber: '',
+    reportCategory: '',
+    otherCategory: '',
+    address: '',
+    gpsCoordinates: '',
+    issueDescription: '',
+    attachments: '',
+    email: false,
+    sms: false,
+    statusOfReport: 'Submitted',
+    dateTimeOfSubmission: new Date().toLocaleString('en-US'),
+});
 
 export type ReportEditProps = {
-    report: Omit<IReport, 'reportId'> | undefined;
-    setReport: (newReport: IReport) => void;
+    report: IReport | undefined;
     submitLabel: string;
-    isSubmitLoading: boolean;
-    onSubmit: () => void;
+    onCreate?: (report: NewReport) => void;
+    onEdit?: (report: IReport) => void;
     cancelHref: string;
 };
 
 /** Report editor, used for create and edit flows */
 export default function ReportEdit({
     report,
-    setReport,
     submitLabel,
-    isSubmitLoading,
-    onSubmit,
+    onCreate,
+    onEdit,
     cancelHref,
 }: ReportEditProps) {
-    const updateReport = useCallback(
-        (newValues: Partial<IReport>) => {
-            setReport({
-                ...(report as IReport),
-                ...newValues,
-            });
-        },
-        [report, setReport]
-    );
+    const initializeValues = (): InitialValuesType => {
+        if (isUndefined(report)) {
+            return getInitialReport();
+        }
+        return report!;
+    };
+
+    const handleSubmit = (
+        report: InitialValuesType,
+        formikHelpers: FormikHelpers<InitialValuesType>
+    ) => {
+        const { setSubmitting } = formikHelpers;
+        try {
+            if (onEdit && 'reportId' in report) {
+                onEdit(report as IReport);
+            } else if (onCreate) {
+                onCreate(report as NewReport);
+            }
+        } catch (error) {
+            console.log('Failed to create report', error);
+        } finally {
+            setSubmitting(false);
```

Comment:
Do we need a corresponding `setSubmitting(true)` at the beginning of this function?

We might also need to add `await` to the `onEdit` and `onCreate` so that `setSubmitting(false)` is only called when the submission is actually complete